From 92995debedd542912a506b51203c332ecf3a2829 Mon Sep 17 00:00:00 2001
From: Kishon Vijay Abraham I <kishon@ti.com>
Date: Mon, 2 Jul 2012 12:20:24 +0530
Subject: [PATCH 18/38] usb: phy: fix error handling in usb_get_phy

spin_unlock_irqrestore() was not being called in the error path of
usb_get_phy. It's fixed here.

Reported-by: Marc Kleine-Budde <mkl@pengutronix.de>
Signed-off-by: Kishon Vijay Abraham I <kishon@ti.com>
Signed-off-by: Felipe Balbi <balbi@ti.com>
Signed-off-by: Ajay Kumar Gupta <ajay.gupta@ti.com>
---
 drivers/usb/otg/otg.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/usb/otg/otg.c b/drivers/usb/otg/otg.c
index 88d62b1..1bf60a2 100644
--- a/drivers/usb/otg/otg.c
+++ b/drivers/usb/otg/otg.c
@@ -98,11 +98,12 @@ struct usb_phy *usb_get_phy(enum usb_phy_type type)
 	if (IS_ERR(phy)) {
 		pr_err("unable to find transceiver of type %s\n",
 			usb_phy_type_string(type));
-		return phy;
+		goto err0;
 	}
 
 	get_device(phy->dev);
 
+err0:
 	spin_unlock_irqrestore(&phy_lock, flags);
 
 	return phy;
-- 
1.7.7.6

